code/__HELPERS/ai.dm

```diff
@@ -0,0 +1,34 @@
+//Returns a node that is in the direction of this node; must be in the src's adjacent node list
+/obj/effect/ai_node/proc/GetNodeInDirInAdj(dir)
+
+	if(!datumnode.adjacent_nodes || !datumnode.adjacent_nodes.len)
+		return null
+
+	for(var/obj/effect/ai_node/node in datumnode.adjacent_nodes)
```

Comment:
Are there non nodes in the adjacent_nodes list?

Reply:
There should never be non nodes in the adjacent nodes list

Reply:
Then make it a typeless loop, 
```
for(var/i in datumnode.adjacent_nodes)
	var/obj/effect/ai_node/node = i
```

Reply:
Implemented